### PR TITLE
refactor: add integration config for bottle, refactor distributed tracing helper usage

### PR DESCRIPTION
**Pull request by Yun-Kim**
## Description
From #2223, it was found that the `bottle` integration did not declare its own integration config and instead relied on storing `distributed_tracing` in the app object. To unify integrations to follow the same pattern, an integration config was added to bottle with the `distributed_tracing` setting set to True by default, and changing the attribute itself to have getter/setters as a property within the app class. Additionally, the activate_distributed_headers helper from trace_utils was used to refactor distributed tracing in bottle.

### CODE REVIEW

File: ddtrace/contrib/bottle/trace.py [hunk 1]
```diff
@@ -11,17 +11,26 @@
...
 from ...ext import SpanTypes
 from ...propagation.http import HTTPPropagator
+from ...utils.formats import asbool
 
 
 class TracePlugin(object):
     name = "trace"
     api = 2
 
     def __init__(self, service="bottle", tracer=None, distributed_tracing=True):
```
Comment (Kyle-Verhoog) [inline ddtrace/contrib/bottle/trace.py:21]:
```suggestion
    def __init__(self, service="bottle", tracer=None, distributed_tracing=None):
```

File: ddtrace/contrib/bottle/trace.py [hunk 2]
```diff
@@ -11,17 +11,26 @@
...
-        self.service = ddtrace.config.service or service
+        self.service = config.service or service
         self.tracer = tracer or ddtrace.tracer
         self.distributed_tracing = distributed_tracing
```
Comment (Kyle-Verhoog) [inline ddtrace/contrib/bottle/trace.py:24]:
```suggestion
        if distributed_tracing is not None:
            config.bottle.distributed_tracing = distributed_tracing
```

File: ddtrace/contrib/bottle/trace.py [hunk 3]
```diff
@@ -11,17 +11,26 @@
...
 
+    @property
+    def distributed_tracing(self):
+        return config.bottle.get("distributed_tracing", True)
```
Comment (Kyle-Verhoog) [inline ddtrace/contrib/bottle/trace.py:28]:
```suggestion
        return config.bottle.distributed_tracing
```

The default is included in the config definition

File: tests/contrib/bottle/test_distributed.py [hunk 4]
```diff
@@ -30,11 +30,13 @@ def tearDown(self):
...
     def _trace_app_distributed(self, tracer=None):
+        # DD_BOTTLE_DISTRIBUTED_TRACING enabled by default
         self.app.install(TracePlugin(service=SERVICE, tracer=tracer))
         self.app = webtest.TestApp(self.app)
 
     def _trace_app_not_distributed(self, tracer=None):
-        self.app.install(TracePlugin(service=SERVICE, tracer=tracer, distributed_tracing=False))
+        ddtrace.config.bottle["distributed_tracing"] = False
+        self.app.install(TracePlugin(service=SERVICE, tracer=tracer))
         self.app = webtest.TestApp(self.app)
```
Comment (Kyle-Verhoog) [inline tests/contrib/bottle/test_distributed.py:41]: Just need a test case that uses the DD_BOTTLE_DISTRIBUTED_TRACING environment variable

**Review by Kyle-Verhoog — CHANGES_REQUESTED** [review]
Just one test case with the env var and this is good to go

**Review by majorgreys — APPROVED** [review]

**Pull request merged**